### DISCUSSION TITLE
Unable to parse string identifier

### DISCUSSION THREAD

**Discussion by kaf-lamed-beyt** [community/community#24752] [Apps, API and Webhooks]
<p>I’m currently trying to make a POST http request to the GitHub GraphQL API, by retrieving the value from the input field and pass it into the GraphQL query that I’ve written but I keep getting this particular error below:</p>
<pre><code class="lang-auto">message: "Parse error on "user" (IDENTIFIER) at [2, 21]
</code></pre>
<p>Here’s the query that I wrote and the fetch API call below</p>
<pre><code class="lang-auto">const getUserRepo = (queryString) =&gt; `
    query userRepos(${queryString}: String!) {
        user(login: "${queryString}") {
            avatarUrl
            name
            bio
            login
            repositories(first: 20) {
              nodes {
                name
                url
                createdAt
                stargazerCount
                forkCount
                description
                primaryLanguage {
                  name
                  color
                }
              }
            }
        }
    }
`;
</code></pre>
<pre><code class="lang-auto">const fetchResults = (e) =&gt; {
  e.preventDefault();

  const searchTerm = form.elements["search"].value;

  const queryMethod = {
    method: "post",
    headers: {
      "Content-Type": "application/json",
      Authorization: "bearer " + api_token,
      "User-Agent": "repo-search-client",
    },
    body: JSON.stringify({
      query: getUserRepo(searchTerm),
    }),
  };

  fetch("https://api.github.com/graphql", queryMethod)
    .then((res) =&gt; res.json())
    .then((data) =&gt; console.log(data))
    .catch((err) =&gt; console.log(JSON.stringify(err)));

  form.reset();
};
</code></pre>
<p>I’d be glad if anyone could render any assitance. Thanks</p>

**Comment by kaf-lamed-beyt** [comment, marked as answer]
<p>I fixed it already.</p>
<p>The error came from me using <code>query userRepos(${queryString}: String!) { }</code> instead of</p>
<pre><code class="lang-auto">const getUserRepo = (queryString) =&gt; `
    query {
        user(login: "${queryString}") {
            avatarUrl
            name
            bio
            login
            repositories(first: 20) {
              nodes {
                name
                url
                createdAt
                stargazerCount
                forkCount
                description
                primaryLanguage {
                  name
                  color
                }
              }
            }
        }
    }
`;
</code></pre>

**Reply by skeggse** [reply]
That's not good practice, for the posterity. Notably if `queryString` is untrusted user input. GraphQL is definitely designed to correctly interpolate values, and you can do that [by passing in variables](https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#working-with-variables).
